main.tf

```diff
@@ -221,6 +223,27 @@ resource "aws_s3_bucket_logging" "default" {
   target_bucket = var.logging.target_bucket
   target_prefix = var.logging.target_prefix
 
+  dynamic "target_object_key_format" {
```

Comment:
In this module we tent to use

var.logging.target_object_key_format != null ?  { create = true } : {}

Reply:
same for the other dynamics

Reply:
var.logging[0].target_object_key_format != null ? [var.logging[0].target_object_key_format] : [] 

Why are you using [0] to reference the variable? I don't think thats needed. And is there a reason for not using something like 

var.logging.target_object_key_format != null ? { create = true } : {} ? that's how we do it at other places in this module and I like consistency :-)

Reply:
Fixed the [0] 

i can't do the {create: true} here, but it actually needs to pass the value from the for_each